File: 101_symmetric_tree.py
from package import TreeNode
class Solution:
    def isSymmetric(self, root):
        if root is None:
            return True
        return self.helper(root.left, root.right)
    def helper(self, p, q):
        if p and q:
            return p.val == q.val and self.helper(p.left, q.right) and self.helper(p.right, q.left)
        return p is q

File: 101_symmetric_tree.cpp
#include <bits/stdc++.h>
#include "package.h"
using namespace std;

class Solution{
    public:
        bool isSymmetric(TreeNode* root){
            if(!root)
                return true;
            return self.isSymmetric(root->left, root->right);
        }
        bool isSymmetric(TreeNode* p, TreeNode* q){
            if(p && q)
                return p->val == q->val && isSymmetric(p->left, q->right) &&
                    isSymmetric(p->right, q->left);
            return p == q;
        }
};
